### PR TITLE
(docs)(GH-1802) Rework modules pages

### LINKED ISSUE #1802: Rework modules pages
- [x] Move modules out of config section
- [x] Change page title: "Module directory structure" to "Module structure"
- [x] Module structure is a sub-page of modules
- [x] Modules needs a more comprehensive intro
People use modules to:
   1.  install content from the forge
   2. write and share their own modules
- [x] Instructions on installing modules needs to be more prominent
- [x] The instructions on installing need to be clearer about moving your personal modules to `site-modules` before you add a line to the puppetfile
- [x] Both modules pages need a general edit

### PR DESCRIPTION
**Pull request by hestonhoffman**
- Move modules to a new section
- Change page title: "Directory structure" to "Module structure"
- Renamed file "directory_structure.md to "module_structure.md"
  - I'll add a redirect for this to the web repo
- Change page title: "Modules" to "Installing modules"
- Updated Installing modules instructions

Closes #1802 
!no-release-note

### CODE REVIEW

File: documentation/bolt_installing_modules.md [hunk 1]
```diff
@@ -49,36 +86,10 @@ This list of packaged modules is available in a [Puppetfile](https://github.com/
...
-    mod 'puppetlabs-haproxy', git: 'https://github.com/puppetlabs/puppetlabs-haproxy.git', ref: 'master'
-    ```
-1.  Add any task or plan modules stored locally in `modules/` to the list. If these modules are not listed in the Puppetfile, they are deleted.
-    ```
-    mod 'my_awesome_module', local: true
-    ```
-    Install any modules that you don't want to manage with the Puppetfile to a different subdirectory. Conventionally, `site-modules` is used for this purpose and is included on the default `modulepath`.
-1.  From a terminal, install the modules listed in the Puppetfile: `bolt puppetfile install`. Bolt installs modules to the first directory in the modulepath setting. By default, this is the `modules/` subdirectory inside the Bolt project directory. To override this location, update the modulepath setting in the Bolt config file.
+📖 **Related information**  
+- For modules that require Ruby Gems, see [Install Gems with Bolt packages](bolt_installing.md#)
```
Comment (lucywyman) [inline documentation/bolt_installing_modules.md:90]:
```suggestion
- For modules that require Ruby Gems, see [Install Gems with Bolt packages](bolt_installing.md#install-gems-with-bolt-packages)
```

Comment (lucywyman) [reply documentation/bolt_installing_modules.md:90]: I also wonder if we should rename that section to "Install gems in Bolt's ruby environment" or something?

Comment (hestonhoffman) [reply documentation/bolt_installing_modules.md:90]: Yeah, that makes a lot of sense. I'll push up a change for that

File: documentation/bolt_installing_modules.md [hunk 2]
```diff
@@ -1,14 +1,51 @@
...
+To share the Bolt plans and tasks that you've created on the Forge, you can
+package them into Puppet modules. If you'd like to use a plan or task from a
+module that you found on the Forge, you can use Bolt to install the module.
+
+## Install a module
+
+> **Before you begin**
+> 
+> - In your Bolt project directory, create a file named `Puppetfile`. 
+> - Add any task or plan modules stored locally in `modules/` to the list. For
```
Comment (hestonhoffman) [inline documentation/bolt_installing_modules.md:12]:
```suggestion
> - Add any task or plan modules stored locally in `modules/` to the list. For
```
@lucywyman I just noticed this. Do you think it's better to say "Add any tasks or plans stored in..." or "Add any module content stored in..."

Comment (lucywyman) [reply documentation/bolt_installing_modules.md:12]: Maybe just 'Add any modules stored locally' or 'Add any Puppet modules stored locally'? Now that plan and task content can be at the top-level directory, I don't expect as many people to get started with Bolt by manually creating a module in `modules/`.

File: documentation/bolt_installing_modules.md [hunk 3]
```diff
@@ -1,14 +1,51 @@
...
+>   example, 
+>   ```puppet
+>     mod 'my_awesome_module', local: true
+>   ```
+> 
+>   **Bolt deletes any content in `modules/` that is not listed in your
+>   Puppetfile.** If you
+>   want to keep the content, but you don't want to manage it with the Puppetfile,
+>   move the content to a `site-modules` directory in your project.
```
Comment (beechtom) [inline documentation/bolt_installing_modules.md:21]: I believe all of the content in `modules/` will be deleted before installing the modules in the Puppetfile. So even if `modules/` contained only the modules in the Puppetfile, any changes you made to those modules will be deleted when running `bolt puppetfile install`.

Comment (lucywyman) [reply documentation/bolt_installing_modules.md:21]: I actually checked this (cause I thought the same) and it's not! If you modify a module that's in the Puppetfile, if doesn't look to get deleted.

Comment (beechtom) [reply documentation/bolt_installing_modules.md:21]: That's weird. I also checked and it removes any changes I made, whether it's new files/directories or changes to existing files.

Comment (beechtom) [reply documentation/bolt_installing_modules.md:21]: Huh, it might just be the `facts` module that that applies to...

**Pull request merged**